src/main/java/eiteam/esteemedinnovation/naturalphilosophy/ItemResearchLog.java

```diff
@@ -2,27 +2,29 @@
 
 import net.minecraft.client.gui.GuiScreen;
 import net.minecraft.client.resources.I18n;
+import net.minecraft.client.util.ITooltipFlag;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.nbt.NBTTagList;
 import net.minecraft.nbt.NBTTagString;
 import net.minecraft.util.text.TextFormatting;
+import net.minecraft.world.World;
 import net.minecraftforge.common.util.Constants;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import javax.annotation.Nullable;
 import java.util.List;
 
 public class ItemResearchLog extends Item {
     public ItemResearchLog() {
         setMaxStackSize(1);
     }
 
-    @SideOnly(Side.CLIENT)
     @Override
-    public void addInformation(ItemStack stack, EntityPlayer player, List<String> tooltip, boolean advanced) {
+    public void addInformation(ItemStack stack, @Nullable World worldIn, List<String> tooltip, ITooltipFlag flagIn) {
```

Comment:
Hm. I should probably specify in the code style to not do Forge's annoying `In` parameter suffixes because they annoy the living shit out of me. Not going to require you to change it because it's not an explicitly established style. This is more of a note to myself.

Reply:
I hate them too. I just generated the methods without touching the variable names.